src/category_theory/limits/shapes/products.lean

```diff
@@ -191,4 +191,69 @@ abbreviation has_products := Π (J : Type v), has_limits_of_shape (discrete J) C
 /-- An abbreviation for `Π J, has_colimits_of_shape (discrete J) C` -/
 abbreviation has_coproducts := Π (J : Type v), has_colimits_of_shape (discrete J) C
 
+section reindex
+variables {C} {γ : Type v} (ε : β ≃ γ) (f : γ → C)
+
+section
+variables [has_product f] [has_product (f ∘ ε)]
+
+/-- Reindex a categorical product via an equivalence of the index types. -/
+def pi.reindex : pi_obj (f ∘ ε) ≅ pi_obj f :=
+has_limit.iso_of_equivalence (discrete.equivalence ε) (discrete.nat_iso (λ i, iso.refl _))
+
+@[simp, reassoc]
+lemma pi.reindex_hom_π (b : β) : (pi.reindex ε f).hom ≫ pi.π f (ε b) = pi.π (f ∘ ε) b :=
+begin
+  dsimp [pi.reindex],
+  simp only [has_limit.iso_of_equivalence_hom_π, discrete.nat_iso_inv_app,
+    equivalence.equivalence_mk'_counit, discrete.equivalence_counit_iso, discrete.nat_iso_hom_app,
+    eq_to_iso.hom, eq_to_hom_map],
+  dsimp,
+  simpa using limit.w (discrete.functor (f ∘ ε)) (eq_to_hom (ε.symm_apply_apply b)),
+end
+
+@[simp, reassoc]
+lemma pi.reindex_inv_π (b : β) : (pi.reindex ε f).inv ≫ pi.π (f ∘ ε) b = pi.π f (ε b) :=
+begin
+  dsimp [pi.reindex],
```

Comment:
Can you get this version from the previous without needing to break down `pi.reindex`?

Reply:
Good call.